Slime Mutation Changes +new reactions

KorPhaeron opened the pull request:
Slime mutation chance starts at a flat 30%

Slime mutation chance is now hereditary, with the new baby slimes
inheriting the mutation chance of the parent (with a variation of -5 to
+5). This allows you to cultivate slimes that are more likely to mutate
over time rather than praying to the RNG forever.

Plasma no longer causes slimes to have a higher mutation chance, and
epinephrine no longer lowers mutation chance.

Red slime extract+plasma now gives a potion that increases mutation
chance. Slime glycerol is no longer obtainable but I don't think I've ever seen anyone use it anyway.

Blue slime extract+blood now gives a potion that lowers mutation chance.

The numbers might need tweaking if this makes xenobio too fast, but
right now it is often frustrating and slow. It is more reliable and
faster to build a mech than breed slimes given the current mutation
system. Hopefully red and blue slimes will be useful now as well.

No Sepia slime fix yet, sorry.

imtakingabreakdontatme committed:
Slime Mutation Changes +new reactions

Slime mutation chance starts at a flat 27%

Slime mutation chance is now hereditary, with the new baby slimes
inheriting the mutation chance of the parent (with a variation of -5 to
+5). This allows you to cultivate slimes that are more likely to mutate
over time rather than praying to the RNG forever.

Plasma no longer causes slimes to have a higher mutation chance, and
epinephrine no longer lowers mutation chance.

Red slime extract+plasma now gives a potion that increases mutation
chace

Blue slime extract+blood now gives a potion that lowers mutation chance.

The numbers might need tweaking if this makes xenobio too fast, but
right now it is often frustrating and slow. It is more reliable and
faster to build a mech than breed slimes given the current mutation
system.

RemieRichards commented on code/modules/mob/living/simple_animal/slime/life.dm:
```diff
@@ -147,7 +147,7 @@
 	return temp_change
 
-/mob/living/simple_animal/slime/handle_chemicals_in_body()
+/*/mob/living/simple_animal/slime/handle_chemicals_in_body()
```
don't comment out code, just removed it, git stores backups should we ever want to revert.

KorPhaeron replied:
Old habit from google code, sorry

RemieRichards replied:
Yeah it's no biggy, I figured that was your reasoning, no worries.

Anonus commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -256,6 +263,58 @@
+
+/obj/item/weapon/slimestabilizer/attack(mob/living/simple_animal/slime/M as mob, mob/user as mob)
+	if(!isslime(M))
+		user << "<span class='warning'> The stabilizer only works on baby slimes!</span>"
```
It's not just baby slimes though

KorPhaeron replied:
Whoops. Not sure how I managed that. I mean I know what I did but still.

Anonus commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -256,6 +263,58 @@
+
+/obj/item/weapon/slimemutator/attack(mob/living/simple_animal/slime/M as mob, mob/user as mob)
+	if(!isslime(M))
+		user << "<span class='warning'> The mutator only works on baby slimes!</span>"
```
See above

RemieRichards commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -256,6 +263,58 @@
+	if(M.stat)
+		user << "<span class='warning'> The slime is dead!</span>"
+		return..()
+	if(M.mutator_used == 1)
```
swap this for ```if(M.mutator_used)``` 

alternatively, make it a proper boolean with
```mutator_used = FALSE```
and then this check remaining ```if(M.mutator_used)

RemieRichards commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -256,6 +263,58 @@
+
+	user <<"<span class='notice'>You feed the slime the mutator. It is now more likely to mutate.</span>"
+	M.mutation_chance += 5
+	if(M.mutation_chance < 0)
```
replace this line, the one above it and the one beneath it with
```M.mutation_chance = max(M.mutation_chance+5,0)```

RemieRichards commented on code/modules/mob/living/simple_animal/slime/powers.dm:
```diff
@@ -191,6 +191,7 @@
 				if(i != 1) step_away(M,src)
 				M.Friends = Friends.Copy()
 				babies += M
+				M.mutation_chance = mutation_chance+(rand(-5,5))
```
```M.mutation_chance = Clamp(mutation_chance+(rand(5,-5)),0,100)```

RemieRichards commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -256,6 +263,58 @@
+		return..()
+
+	user <<"<span class='notice'>You feed the slime the stabilizer. It is now less likely to mutate.</span>"
+	M.mutation_chance -= 15
```
```M.mutation_chance = Clamp(M.mutation_chance,0,100)```

KorPhaeron replied:
M.mutation_chance = Clamp(M.mutation_chance-15,0,100) work?

Or does clamp only do positive values?

RemieRichards replied:
Oh yeah, I meant to include the -15 in the clamp statement.

```Clamp(value, minimum, maximum)```
it puts the number ```value``` between ```minimum``` and ```maximum```
So if you range includes negatives (Clamp(health, -100, 100)) then it handles them.

Man-Shroom commented:
Red/Blue slime extract should have these new effects if combined with water, and keep their old effects.

RemieRichards commented:
Also ```KorPhaeron:master``, you shouldn't use your master branch for Pull requests, it makes your life harder (doesn't affect us at all)

AlexCrimson commented:
I dont think this change will do anything. People are not gonna get red/blue slimes just so they can get some reagent to lower/increase mutation chance. Its easy enough to get the specific slimes you want without dealing with mutation chance.

Just make water lower the chance, and mutagen increase it.

Man-Shroom commented:
the above would work too

KorPhaeron commented:
I thought about mutagen, but it's more or less in infinite supply.

AlexCrimson commented:
Thats kinda the point. Making it a hard to get chemical will just make it a useless feature nobody ever uses. Right now Xeno does perfectly fine and never has to inject a single slime with plasma or the other stuff to change mutation chance. Making it an ever rarer chemical sure as hell wont suddenly get people to use it.

KorPhaeron commented:
I wanted the +mutation to be a bit harder to get now that the bonus will persist between generations.

If I made mutagen give +mutation chance they could dump 50 units into the starting slime and all the children would have a crazy mutation chance as well.

AlexCrimson commented:
Nobody will bother with Red/Blue slimes just so they can mutate slimes faster. It would take less time to aim for certain slimes compared to getting Red slimes and engineering slimes with high mutation chance.

If its what you want to do then fine, but nobody is going to use it.

KorPhaeron commented:
I think you're correct after thinking further on it. I usually try and err on making my stuff underpowered but I think I may have made the values too low (both starting flat chance and the mutator bonus) for the new system to actually be helpful. I'll tweak it upwards to make red slimes an attractive alternative to try for if you happen to get orange on the first split.

AlexCrimson commented:
Eh that would probably be a good solution. Still think the reaction should only require water though. Plasma seems too valuable to waste converting into this stuff.

imtakingabreakdontatme committed:
More slime stuff

Further tweaks slime mutation chances based on Crimson's feedback, makes
the code prettier based on Remie's feedback

imtakingabreakdontatme committed:
More changes

Based on Remie's suggestions

imtakingabreakdontatme committed:
Update slime.dm

Fixed mutator used to be boolean
